### ISSUE TITLE
Output bam not complete

### ISSUE THREAD

**Issue by Glfrey** [mhalushka/miRge3.0#77]
Hello again!

I'm trying to output results in sam or bam format, but the command results in a bam file which when converted contains only headers, with no sequence. Stopping mirge early shows that sam files are being created, filled with alignment information, but wherever the merge is happening into a single output bam file (and the individual sam files deleted) the actual alignment information isn't being carried over.

I'm trying to look at it myself, but I can't seem to find the culprit. When you get a moment, could you take a look please?

Thank you!

**Comment by arunhpatil** [comment]
Hi @Glfrey, 

This is interesting, can you share the command you used to convert BAM to SAM? I have the files and work environment on an another machine and I will check this and get back to you.

Thank you,
Arun.

**Comment by Glfrey** [comment]
Hi @arunhpatil,

Sure, I used` samtools view -h -o out.sam in.bam`. 

Thank you!

Gill

**Comment by arunhpatil** [comment]
Hi @Glfrey,

I checked the files and it is showing the sequences in both BAM and SAM format. Here is what I did:

```
Pseudo-script:
samtools view SRR772403.bam | tail -5
samtools view -h -o  out.sam  SRR772403.bam
tail -5 out.sam
```

The output of the above code:
```

(base) arun@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ samtools view miRge.2023-07-10_16-16-59/SRR772403.bam | tail -5
TAGTAGGTTGTATAGTTTGGGGCTCTGT_0  0       chr22   46112757        255     28M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGT    IIIIIIIIIIIIIIIIIIIIIIIIIIII        XA:i:1  MD:Z:27C0       NM:i:1  XM:i:1
CGGGGATGACTGTGACCACATTGGGATGTATTCGTACTGTCTGATG_0        16      chr19   12703601        255     46M     *       0       0       CATCAGACAGTACGAATACATCCCAATGTGGTCACAGTCATCCCCG      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:46 NM:i:0  XM:i:1
TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT_0    0       chr22   46112757        255     34M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:34 NM:i:0  XM:i:1
CCGGGGATGACTGTGACCACATTGGGATGTATTCGTCCTGTCTGATG_0       16      chr19   12703601        255     47M     *       0       0       CATCAGACAGGACGAATACATCCCAATGTGGTCACAGTCATCCCCGG     IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII XA:i:0  MD:Z:36A10      NM:i:1  XM:i:1
CATTGGGATGTATTCGTACTGTCTGATG_0  16      chr19   12703601        255     28M     *       0       0       CATCAGACAGTACGAATACATCCCAATG    IIIIIIIIIIIIIIIIIIIIIIIIIIII        XA:i:0  MD:Z:28 NM:i:0  XM:i:1


(base) arun@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ samtools view -h -o out.sam miRge.2023-07-10_16-16-59/SRR772403.bam


(base) arun@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ tail -5 out.sam
TAGTAGGTTGTATAGTTTGGGGCTCTGT_0  0       chr22   46112757        255     28M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGT    IIIIIIIIIIIIIIIIIIIIIIIIIIII        XA:i:1  MD:Z:27C0       NM:i:1  XM:i:1
CGGGGATGACTGTGACCACATTGGGATGTATTCGTACTGTCTGATG_0        16      chr19   12703601        255     46M     *       0       0       CATCAGACAGTACGAATACATCCCAATGTGGTCACAGTCATCCCCG      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:46 NM:i:0  XM:i:1
TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT_0    0       chr22   46112757        255     34M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:34 NM:i:0  XM:i:1
CCGGGGATGACTGTGACCACATTGGGATGTATTCGTCCTGTCTGATG_0       16      chr19   12703601        255     47M     *       0       0       CATCAGACAGGACGAATACATCCCAATGTGGTCACAGTCATCCCCGG     IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII XA:i:0  MD:Z:36A10      NM:i:1  XM:i:1
CATTGGGATGTATTCGTACTGTCTGATG_0  16      chr19   12703601        255     28M     *       0       0       CATCAGACAGTACGAATACATCCCAATG    IIIIIIIIIIIIIIIIIIIIIIIIIIII        XA:i:0  MD:Z:28 NM:i:0  XM:i:1
(base) arun@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$
```

Can you check the miRNA summary file? Most likely, there are no mapping probably due to incorrect adaptor trimming. 

Thank you,
Arun.

**Comment by Glfrey** [comment]
Hi @arunhpatil ,

That's interesting. I actually do have mappings, as I'm getting alignments in the intermediate sam files. Interestingly though, according to the summary I have 0 input reads, but still have annotations. All I can think of is if my changes to mirge have caused any issues with read counting and sam to bam conversion? Does the version you tested on contain my changes? 

<img width="1398" alt="Screenshot 2023-07-11 at 09 13 25" src="https://github.com/mhalushka/miRge3.0/assets/33197909/69c7a211-23be-4ec1-9ef5-e0db9d3b5563">

**Comment by Glfrey** [comment]
A rerun of the data without the `--miREC ` parameter gives me input read counts back, but no adjustment of the parameters is giving me content in the bam file, so my guess is it's something to do with the changes made on my end. If you were running mirge with the updates I made and still have content in the bam file then it's something else I've implemented on my own repository that's caused it. I'll go back a few changes and see what happens. 

<img width="1400" alt="Screenshot 2023-07-11 at 10 40 39" src="https://github.com/mhalushka/miRge3.0/assets/33197909/0a38dcff-38dc-4976-b7ed-e899265463ac">

**Comment by Glfrey** [comment]
So I've done some digging and I've identified a potential culprit but it doesn't explain why you get results. Looking through `summary.py', it looks like the bam file that's empty of alignment information for me only ever has headers written to it:

```
with open(sam_nameH,"w+") as samH:
                #samH.write("@HD\tVN:3.0\tSO:coordinate\n")
                samH.write(header)
```
A search of `sam_nameH` through summary.py shows it only present in these locations, so the file is never reopened and anything else written to it.

```
file_sam_nameH = str(names) +".sam"
            sam_nameH = Path(workDir)/file_sam_nameH
            with open(sam_nameH,"w+") as samH:
                #samH.write("@HD\tVN:3.0\tSO:coordinate\n")
                samH.write(header)
 
```

This explains why nothing is being written into my file but not why something is being written into yours. I didn't make any changes to the `summary.py` file at any point so I'm at a loss to explain it. Are you perhaps working with a version from over a year ago (when I can see changes were made to `summary.py` or am I missing the file being recalled somewhere else?

**Comment by Glfrey** [comment]
Hi @arunhpatil ,

Sorry for spamming the chat, but I think I've identified the problem. Using print statements as a debugging tool I've identified that nothing is being written into the core sam file (e.g. sampleB.sam). In the function `bow2bam` the code isn't progressing past where I've put "here1" (ignore the "S>" symbol it's just an artefact of my code viewer). As such, no further processing of the sam file is happening and nothing is being written out.

I've had issues in the past with code following an if then else statement not being executed in python when within a "try except" block and I think that is what is happening here. 

```
with open(otrna_samFile) as miSam:
          for mi_idx, mi_sam in enumerate(miSam):
              mi_sam = mi_sam.strip()
              mi_sam_list = mi_sam.split("\t")
              try:
S>                sam_exprn_list = bam_expression_dict[mi_sam_list[0]] # list of expression values 
                  try:
                      if args.organism_name == "hamster":
S>                        chromo = str(dict_genCors[mi_sam_list[2]].split(":")[2])
                      else:
                          print("here1")
                          chromo = str(dict_genCors[mi_sam_list[2]])
S>                        # "chr"+ str(dict_genCors[mi_sam_list[2]].split(":")[2])
S>                        #if chromo == "chrMT": 
                          #    chromo = chromo.replace("chrMT", "chrM")
S>                    # start = int(dict_genCors[mi_sam_list[2]].split(":")[3]) + int(mi_sam_list[3]) - 1
                      strand = str(dict_genCors[mi_sam_list[2]+"#strand"])
                      strand = str(dict_genCors[mi_sam_list[2]+"#strand"])
                      print("here2")
                      if strand == "+":
                          print("here3")
S>                        start, cigar = fetch_pos_coordinate(mi_sam_list[3], dict_genCors[mi_sam_list[2]+"#segs"], dict_genCors[mi_sam_list[2]+"#cds"], len(mi_sam_list[9]))
                          mi_sam_list[1] = "0"
                          sequence = mi_sam_list[9]
S>                        # Returned CIGAR is 0 or spliced alignment 
                          if cigar == 0:
S>                            #cigar = str(len(mi_sam_list[8]))+"M"
                              cigar = str(mi_sam_list[5])
                      else:
                          print("here4")
S>                        start, cigar = fetch_neg_coordinate(mi_sam_list[3], dict_genCors[mi_sam_list[2]+"#segs"], dict_genCors[mi_sam_list[2]+"#cds"], len(mi_sam_list[9]))
                          mi_sam_list[1] = "16"
                          sequence = str(Seq(mi_sam_list[9][::-1]).complement())
                          if cigar == 0:
                              cigar = str(mi_sam_list[5])
                      print("here!")
                      print("trying to write")
                      for ex_idx, exprn in enumerate(sam_exprn_list):
                          print("ex_idx")
                          print(ex_idx)
                          print("exprn")
                          print(exprn)



```

I double checked my theory by placing a print statement just below the else, but with the same indentation as the following code and I can confirm nothing is being executed after the "else" statement (i.e. print("here here here") isn't being printed).

```
with open(otrna_samFile) as miSam:
          for mi_idx, mi_sam in enumerate(miSam):
              mi_sam = mi_sam.strip()
              mi_sam_list = mi_sam.split("\t")
              try:
S>                sam_exprn_list = bam_expression_dict[mi_sam_list[0]] # list of expression values 
                  try:
                      if args.organism_name == "hamster":
S>                        chromo = str(dict_genCors[mi_sam_list[2]].split(":")[2])
                      else:
                          print("here1")
                          chromo = str(dict_genCors[mi_sam_list[2]])
S>                        # "chr"+ str(dict_genCors[mi_sam_list[2]].split(":")[2])
S>                        #if chromo == "chrMT": 
                          #    chromo = chromo.replace("chrMT", "chrM")
S>                    # start = int(dict_genCors[mi_sam_list[2]].split(":")[3]) + int(mi_sam_list[3]) - 1
                      print("here here here")
                      
                      strand = str(dict_genCors[mi_sam_list[2]+"#strand"])
                      print("here2")
                      if strand == "+":
                          print("here3")
```

I'm trying to remember how I solved this issue in the past, I'll need to go check my repositories. I'll get back to you when I find a solution.

**Comment by arunhpatil** [comment]
Hi @Glfrey,

I have not got a chance to go through the issues, but I will get back to you on this soon. 

Thank you,
Arun.

**Comment by Glfrey** [comment]
Hi @arunhpatil,

No hurry, I'm at a conference until Sunday evening so I'll be limited in my ability to reply until then anyway.

**Comment by arunhpatil** [comment]
Hi @Glfrey,

TL;DR: The BAM files are fine. 

I created two new environment (original and glfrey) and installed miRge3.0 in both. Next, I replaced the `manifoldAlign.py` in glfrey environment and then run a sample file in both. Both the environments produced same BAM format. Something has changed in your clone/fork. I hope this helps. 

```
(glfrey) apatil8@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ miRge3.0 -s SRR772403.fastq.gz -a illumina -bam -lib /mnt/d/Halushka_lab/Arun/GTF_Repeats_miRge2to3/miRge3_Lib/revised_hsa -on human -db mirbase -o output_dir/
/home/apatil8/miniconda3/envs/glfrey/lib/python3.10/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.
  warnings.warn(
bowtie version: 1.3.1
cutadapt version: 4.4
Samtools version: 1.17
Collecting and validating input files...

miRge3.0 will process 1 out of 1 input file(s).

Cutadapt finished for file SRR772403 in 2.0165 second(s)
Collapsing finished for file SRR772403 in 0.0114 second(s)

Matrix creation finished in 0.0165 second(s)

Data pre-processing completed in 2.0738 second(s)

Alignment in progress ...
Alignment starting for exact miRNA...
Number of molecules with valid alternate alignments for exact miRNA:0
Alignment starting for hairpin miRNA...
Number of molecules with valid alternate alignments for hairpin miRNA:0
Alignment starting for mature tRNA...
Number of molecules with valid alternate alignments for mature tRNA:845
 This accounts for 74.3% of all aligned molecules
 See output_dir/miRge.2023-07-15_15-13-21/mature_tRNA_secondary_matches.csv for alignment details
Alignment starting for primary tRNA...
Number of molecules with valid alternate alignments for primary tRNA:0
Alignment starting for snoRNA...
Number of molecules with valid alternate alignments for snoRNA:0
Alignment starting for rRNA...
Number of molecules with valid alternate alignments for rRNA:0
Alignment starting for ncrna others...
Number of molecules with valid alternate alignments for ncrna others:0
Alignment starting for mRNA...
Number of molecules with valid alternate alignments for mRNA:0
Alignment starting for isomiR miRNA...
Number of molecules with valid alternate alignments for isomiR miRNA:0
Alignment completed in 132.4446 second(s)

Summarizing and tabulating results...
Summary completed in 114.626 second(s)


The path to ourput directory: /mnt/d/Halushka_lab/Arun/datasets/output_dir/miRge.2023-07-15_15-13-21

The analysis completed in 250.1112 second(s)

(glfrey) apatil8@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ samtools view output_dir/miRge.2023-07-15_15-13-21/SRR772403.bam | tail -5
TAGTAGGTTGTATAGTTTGGGGCTCTGT_0  0       chr22   46112757        255     28M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGT    IIIIIIIIIIIIIIIIIIIIIIIIIIII    XA:i:1  MD:Z:27C0       NM:i:1  XM:i:1
CGGGGATGACTGTGACCACATTGGGATGTATTCGTACTGTCTGATG_0        16      chr19   12703601        255     46M     *       0       0       CATCAGACAGTACGAATACATCCCAATGTGGTCACAGTCATCCCCG  IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:46 NM:i:0  XM:i:1
CATTGGGATGTATTCGTACTGTCTGATG_0  16      chr19   12703601        255     28M     *       0       0       CATCAGACAGTACGAATACATCCCAATG    IIIIIIIIIIIIIIIIIIIIIIIIIIII    XA:i:0  MD:Z:28 NM:i:0  XM:i:1
CCGGGGATGACTGTGACCACATTGGGATGTATTCGTCCTGTCTGATG_0       16      chr19   12703601        255     47M     *       0       0       CATCAGACAGGACGAATACATCCCAATGTGGTCACAGTCATCCCCGG IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII XA:i:0  MD:Z:36A10      NM:i:1  XM:i:1
TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT_0    0       chr22   46112757        255     34M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII      XA:i:0  MD:Z:34 NM:i:0  XM:i:1
```

```
(glfrey) apatil8@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ samtools view -h -o out_glfrey.sam output_dir/miRge.2023-07-15_15-13-21/SRR772403.bam
(glfrey) apatil8@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ less out_glfrey.sam
(glfrey) apatil8@DESKTOP-H231NS7:/mnt/d/Halushka_lab/Arun/datasets$ tail -5 out_glfrey.sam
TAGTAGGTTGTATAGTTTGGGGCTCTGT_0  0       chr22   46112757        255     28M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGT    IIIIIIIIIIIIIIIIIIIIIIIIIIII    XA:i:1  MD:Z:27C0       NM:i:1  XM:i:1
CGGGGATGACTGTGACCACATTGGGATGTATTCGTACTGTCTGATG_0        16      chr19   12703601        255     46M     *       0       0       CATCAGACAGTACGAATACATCCCAATGTGGTCACAGTCATCCCCG  IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  XA:i:0  MD:Z:46 NM:i:0  XM:i:1
CATTGGGATGTATTCGTACTGTCTGATG_0  16      chr19   12703601        255     28M     *       0       0       CATCAGACAGTACGAATACATCCCAATG    IIIIIIIIIIIIIIIIIIIIIIIIIIII    XA:i:0  MD:Z:28 NM:i:0  XM:i:1
CCGGGGATGACTGTGACCACATTGGGATGTATTCGTCCTGTCTGATG_0       16      chr19   12703601        255     47M     *       0       0       CATCAGACAGGACGAATACATCCCAATGTGGTCACAGTCATCCCCGG IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII XA:i:0  MD:Z:36A10      NM:i:1  XM:i:1
TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT_0    0       chr22   46112757        255     34M     *       0       0       TAGTAGGTTGTATAGTTTGGGGCTCTGCCCTGCT      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII      XA:i:0  MD:Z:34 NM:i:0  XM:i:1
```

Thank you,
Arun.

**Comment by arunhpatil** [comment]
Hi @Glfrey, 

I hope you get a chance to see the comment on the recent PR. In case you have missed, please see this [link](https://github.com/mhalushka/miRge3.0/pull/74#issuecomment-1637152429).

Thank you,
Arun.

**Comment by PatrickJReed** [comment]
Has this issue been addressed, im getting the same behavior.  Empty bam files with a fresh install.

**Comment by PatrickJReed** [comment]
run.log:
/home/ubuntu/.local/bin/miRge3.0 -s A1_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,A2_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,A3_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,A4_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,B1_IR_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,B2_IR_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,B3_IR_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,B4_IR_SMALLRNA_ms_5hrs_CTX_R1.fastq.gz,C1_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,C2_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,C3_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,C4_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,D1_IR_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,D2_IR_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,D3_IR_SMALLRNA_ms_12wks_CTX_R1.fastq.gz,D4_IR_SMALLRNA_ms_12wks_CTX_R1.fastq.gz -lib /home/ubuntu/miRge3_Lib -on mouse -db mirgenedb -o output -bam -cpu 16 -a TGGAATTCTCGGGTGC -g UCUUUCCCUACACGACGCUCUUCCGAUCU -cms 1024
bowtie version: 1.3.0
cutadapt version: 4.4
Samtools version: 1.10
Collecting and validating input files...

miRge3.0 will process 16 out of 16 input file(s).

Cutadapt finished for file A1_SMALLRNA_ms_5hrs_CTX_R1 in 19.6495 second(s)
Collapsing finished for file A1_SMALLRNA_ms_5hrs_CTX_R1 in 0.3401 second(s)
Cutadapt finished for file A2_SMALLRNA_ms_5hrs_CTX_R1 in 37.3229 second(s)
Collapsing finished for file A2_SMALLRNA_ms_5hrs_CTX_R1 in 5.7588 second(s)
Cutadapt finished for file A3_SMALLRNA_ms_5hrs_CTX_R1 in 22.6855 second(s)
Collapsing finished for file A3_SMALLRNA_ms_5hrs_CTX_R1 in 6.6318 second(s)
Cutadapt finished for file A4_SMALLRNA_ms_5hrs_CTX_R1 in 23.9572 second(s)
Collapsing finished for file A4_SMALLRNA_ms_5hrs_CTX_R1 in 8.3894 second(s)
Cutadapt finished for file B1_IR_SMALLRNA_ms_5hrs_CTX_R1 in 21.827 second(s)
Collapsing finished for file B1_IR_SMALLRNA_ms_5hrs_CTX_R1 in 9.8204 second(s)
Cutadapt finished for file B2_IR_SMALLRNA_ms_5hrs_CTX_R1 in 23.5228 second(s)
Collapsing finished for file B2_IR_SMALLRNA_ms_5hrs_CTX_R1 in 11.5116 second(s)
Cutadapt finished for file B3_IR_SMALLRNA_ms_5hrs_CTX_R1 in 22.1569 second(s)
Collapsing finished for file B3_IR_SMALLRNA_ms_5hrs_CTX_R1 in 13.7268 second(s)
Cutadapt finished for file B4_IR_SMALLRNA_ms_5hrs_CTX_R1 in 21.6423 second(s)
Collapsing finished for file B4_IR_SMALLRNA_ms_5hrs_CTX_R1 in 15.607 second(s)
Cutadapt finished for file C1_SMALLRNA_ms_12wks_CTX_R1 in 21.8386 second(s)
Collapsing finished for file C1_SMALLRNA_ms_12wks_CTX_R1 in 17.5014 second(s)
Cutadapt finished for file C2_SMALLRNA_ms_12wks_CTX_R1 in 23.2658 second(s)
Collapsing finished for file C2_SMALLRNA_ms_12wks_CTX_R1 in 18.8361 second(s)
Cutadapt finished for file C3_SMALLRNA_ms_12wks_CTX_R1 in 21.1461 second(s)
Collapsing finished for file C3_SMALLRNA_ms_12wks_CTX_R1 in 19.3262 second(s)
Cutadapt finished for file C4_SMALLRNA_ms_12wks_CTX_R1 in 22.9561 second(s)
Collapsing finished for file C4_SMALLRNA_ms_12wks_CTX_R1 in 22.0195 second(s)
Cutadapt finished for file D1_IR_SMALLRNA_ms_12wks_CTX_R1 in 23.0873 second(s)
Collapsing finished for file D1_IR_SMALLRNA_ms_12wks_CTX_R1 in 22.9527 second(s)
Cutadapt finished for file D2_IR_SMALLRNA_ms_12wks_CTX_R1 in 23.5894 second(s)
Collapsing finished for file D2_IR_SMALLRNA_ms_12wks_CTX_R1 in 23.2756 second(s)
Cutadapt finished for file D3_IR_SMALLRNA_ms_12wks_CTX_R1 in 22.8473 second(s)
Collapsing finished for file D3_IR_SMALLRNA_ms_12wks_CTX_R1 in 23.5862 second(s)
Cutadapt finished for file D4_IR_SMALLRNA_ms_12wks_CTX_R1 in 21.2529 second(s)
Collapsing finished for file D4_IR_SMALLRNA_ms_12wks_CTX_R1 in 26.3926 second(s)
Matrix creation finished in 7.8114 second(s)

Data pre-processing completed in 641.7488 second(s)

Alignment in progress ...
Alignment completed in 176.8271 second(s)

Summarizing and tabulating results...
Summary completed in 77.8011 second(s)

The path to ourput directory: /home/ubuntu/23-07b8/output/miRge.2023-08-23_17-53-21
The analysis completed in 901.0541 second(s)

**Comment by PatrickJReed** [comment]
all bams are empty

-rw-rw-r-- 1 ubuntu ubuntu      1732 Aug 23 18:07 A1_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 A1_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1775 Aug 23 18:07 A1_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1732 Aug 23 18:07 A2_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 A2_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1775 Aug 23 18:07 A2_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1732 Aug 23 18:07 A3_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 A3_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1775 Aug 23 18:07 A3_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1732 Aug 23 18:07 A4_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 A4_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1775 Aug 23 18:07 A4_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1736 Aug 23 18:07 B1_IR_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 B1_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1779 Aug 23 18:07 B1_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1736 Aug 23 18:07 B2_IR_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 B2_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1779 Aug 23 18:07 B2_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1736 Aug 23 18:07 B3_IR_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 B3_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1779 Aug 23 18:07 B3_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1736 Aug 23 18:07 B4_IR_SMALLRNA_ms_5hrs_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 B4_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1779 Aug 23 18:07 B4_IR_SMALLRNA_ms_5hrs_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1733 Aug 23 18:07 C1_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 C1_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1777 Aug 23 18:07 C1_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1733 Aug 23 18:07 C2_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 C2_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1777 Aug 23 18:07 C2_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1733 Aug 23 18:07 C3_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 C3_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1777 Aug 23 18:07 C3_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1733 Aug 23 18:07 C4_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 C4_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1777 Aug 23 18:07 C4_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1737 Aug 23 18:07 D1_IR_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 D1_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1780 Aug 23 18:07 D1_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1737 Aug 23 18:07 D2_IR_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 D2_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1780 Aug 23 18:07 D2_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1737 Aug 23 18:07 D3_IR_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 D3_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1780 Aug 23 18:07 D3_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      1737 Aug 23 18:07 D4_IR_SMALLRNA_ms_12wks_CTX_R1.bam
-rw-rw-r-- 1 ubuntu ubuntu       784 Aug 23 18:07 D4_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bai
-rw-rw-r-- 1 ubuntu ubuntu      1780 Aug 23 18:07 D4_IR_SMALLRNA_ms_12wks_CTX_R1_sorted.bam
-rw-rw-r-- 1 ubuntu ubuntu      2330 Aug 23 18:07 annotation.report.csv
-rw-rw-r-- 1 ubuntu ubuntu     48427 Aug 23 18:07 annotation.report.html
-rw-rw-r-- 1 ubuntu ubuntu    163599 Aug 23 18:07 index_data.js
-rw-rw-r-- 1 ubuntu ubuntu 209611788 Aug 23 18:08 mapped.csv
-rw-rw-r-- 1 ubuntu ubuntu     81440 Aug 23 18:07 miR.Counts.csv
-rw-rw-r-- 1 ubuntu ubuntu     97897 Aug 23 18:07 miR.RPM.csv
-rw-rw-r-- 1 ubuntu ubuntu     14519 Aug 23 18:08 miRge3_visualization.html
-rw-rw-r-- 1 ubuntu ubuntu      3807 Aug 23 18:08 run.log
-rw-rw-r-- 1 ubuntu ubuntu 353941300 Aug 23 18:08 unmapped.csv

**Comment by arunhpatil** [comment]
Hi @PatrickJReed 

I couldn't replicate the error with the FASTQ files I have, can you share a sample/subset of the FASTQ file you used? Also, you are trimming linked adapters and you should use the method [trimming-both-5-and-3-adapters-linked-adapters](https://mirge3.readthedocs.io/en/master/quick_start.html#trimming-both-5-and-3-adapters-linked-adapters) for specifying both 5' and 3' adapters. 

Thank you,
Arun
